packages/flutter_tools/lib/src/test/flutter_platform.dart

```diff
@@ -240,57 +239,68 @@ class _Compiler {
-
-          // In case compiler didn't produce output or reported compilation
-          // errors, pass [null] upwards to the consumer and shutdown the
-          // compiler to avoid reusing compiler that might have gotten into
-          // a weird state.
-          if (outputPath == null || compilerOutput.errorCount > 0) {
-            request.result.complete(null);
-            await _shutdown();
-          } else {
-            final File outputFile = fs.file(outputPath);
-            final File kernelReadyToRun = await outputFile.copy('${request.path}.dill');
-            final File testCache = fs.file(testFilePath);
-            if (firstCompile || !testCache.existsSync() || (testCache.lengthSync() < outputFile.lengthSync())) {
-              // The idea is to keep the cache file up-to-date and include as
-              // much as possible in an effort to re-use as many packages as
-              // possible.
-              ensureDirectoryExists(testFilePath);
-              await outputFile.copy(testFilePath);
-            }
-            request.result.complete(kernelReadyToRun.path);
-            compiler.accept();
-            compiler.reset();
+      ResidentCompiler compiler;
+      String outputDill;
+      bool firstCompile = false;
+      if (idleCompilers.isEmpty && compilers < concurrency ) {
+        // Create compiler
+        compiler = createCompiler();
+        firstCompile = true;
+        compilers++;
+        outputDill = outputDillDirectory.childFile('output_$compilers.dill').path;
+        printTrace('Compiler will use the following file as its incremental dill file: $outputDill');
+      } else if (idleCompilers.isEmpty) {
+        // The request was added to the queue, but we can't process it now.
+        return;
+      } else {
+        compiler = idleCompilers.removeLast();
+      }
+      while (compilationQueue.isNotEmpty) {
+        final _CompilationRequest request = compilationQueue.first;
+        // Only remove now when we finished processing the element
+        compilationQueue.removeAt(0);
+        printTrace('Compiling ${request.path}');
+        final Stopwatch compilerTime = Stopwatch()..start();
+        suppressOutput = false;
+        final CompilerOutput compilerOutput = await handleTimeout<CompilerOutput>(
+            compiler.recompile(
+                request.path,
+                <String>[request.path],
+                outputPath: outputDill),
+            request.path);
+        final String outputPath = compilerOutput?.outputFilename;
+
+        // In case compiler didn't produce output or reported compilation
+        // errors, pass [null] upwards to the consumer and shutdown the
+        // compiler to avoid reusing compiler that might have gotten into
+        // a weird state.
+        if (outputPath == null || compilerOutput.errorCount > 0) {
+          request.result.complete(null);
+          await _shutdown(compiler);
```

Comment:
if we get into this path, aren't we supposed to exit the While loop in oder to go back and create a new instance of the compiler, or create a fresh instance of the compiler here?

Reply:
Yes, good point!